stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
+    def make_transitions(
+        self, name_list: list[str | tuple[str] | tuple[str, str]]
+    ) -> tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]: ...
+    def no_match(
+        self, context: list[str], transitions: tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]
+    ) -> _TransitionResult: ...
+    def bof(self, context: list[str]) -> tuple[list[str], list[str]]: ...
+    def eof(self, context: list[str]) -> list[str]: ...
+    def nop(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+
+class StateMachineWS(StateMachine):
+    def get_indented(self, until_blank: bool = False, strip_indent: bool = True) -> tuple[StringList, int, int, bool]: ...
+    def get_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True
+    ) -> tuple[list[str], int, bool]: ...
+    def get_first_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True, strip_top: bool = True
+    ) -> tuple[list[str], int, int, bool]: ...
+
+class StateWS(State):
+    indent_sm: type[StateMachine] | None
+    indent_sm_kwargs: dict[str, Any] | None
+    known_indent_sm: type[StateMachine] | None
+    known_indent_sm_kwargs: dict[str, Any] | None
+    ws_patterns: dict[str, str]
+    ws_initial_transitions: Sequence[str]
+    def __init__(self, state_machine: StateMachine, debug: bool = False) -> None: ...
+    def add_initial_transitions(self) -> None: ...
+    def blank(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def known_indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def first_known_indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+
+class _SearchOverride:
+    def match(self, pattern: Pattern[str]) -> Match[str]: ...
+
+class SearchStateMachine(_SearchOverride, StateMachine): ...
+class SearchStateMachineWS(_SearchOverride, StateMachineWS): ...
+
+_Data = TypeVar("_Data")
+
+class ViewList(Generic[_Data]):
+    data: list[_Data]
+    items: list[tuple[str, int]]
+    parent: ViewList[_Data]
+    parent_offset: int
+    def __init__(
+        self,
+        initlist: ViewList[_Data] | list[_Data] | None = None,
+        source: str | None = None,
+        items: list[tuple[str, int]] | None = None,
+        parent: ViewList[_Data] | None = None,
+        parent_offset: int | None = None,
+    ) -> None: ...
+    def __lt__(self, other: Any) -> bool: ...
+    def __le__(self, other: Any) -> bool: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    def __gt__(self, other: Any) -> bool: ...
+    def __ge__(self, other: Any) -> bool: ...
+    def __contains__(self, item: ViewList[_Data]) -> bool: ...
```

Comment:
this is the implementation:

```python
    def __lt__(self, other): return self.data < self.__cast(other)   # noqa
    def __le__(self, other): return self.data <= self.__cast(other)  # noqa
    def __eq__(self, other): return self.data == self.__cast(other)  # noqa
    def __ne__(self, other): return self.data != self.__cast(other)  # noqa
    def __gt__(self, other): return self.data > self.__cast(other)   # noqa
    def __ge__(self, other): return self.data >= self.__cast(other)  # noqa
```

Reply:
IIRC the `__cast` function is the identity on non-ViewList objects and returns other.data otherwise I think. 

So the expected argument is either a ViewList or a list of data (I think).